rpcs3/Emu/RSX/Common/texture_cache_helpers.h

```diff
@@ -501,6 +503,76 @@ namespace rsx
 			return false;
 		}
 
+		template <typename sampled_image_descriptor>
+		void calculate_sample_clip_parameters(
+			sampled_image_descriptor& desc,
+			const position2i& offset,
+			const size2i& desired_dimensions,
+			const size2i& actual_dimensions)
+		{
+			const f32 scale_x = f32(desired_dimensions.width) / actual_dimensions.width;
+			const f32 scale_y = f32(desired_dimensions.height) / actual_dimensions.height;
+			const f32 offset_x = f32(offset.x) / actual_dimensions.width;
+			const f32 offset_y = f32(offset.y) / actual_dimensions.height;
```

Comment:
what if the dimensions are 0 ?

Reply:
2D regions with dimensions of 0 don't exist.